src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -382,17 +430,145 @@ class ContentExplorer extends Component {
+        return items && items[0] && items[0].isLoading;
+    };
+
+    handleSelectAllClick = async () => {
+        if (this.itemsAreLoading()) {
             return;
         }
         const { isSelectAllChecked } = this.state;
         const newSelectedItems = isSelectAllChecked ? this.unselectAll() : this.selectAll();
         this.setState({ selectedItems: newSelectedItems, isSelectAllChecked: !isSelectAllChecked });
     };
 
+    shouldSelectAndActionDisableItems = () => {
+        const { folderToDeepScan, foldersPath } = this.state;
+        let result = false;
+
+        if (this.itemsAreLoading()) {
+            return result;
+        }
+
+        // Check to see if the folder that was selected for include subfolders is in our current folder path
+        foldersPath.forEach(folder => {
+            if (folder.id === folderToDeepScan.id) {
+                result = true;
+            }
+        });
+
+        return result;
+    };
+
+    shouldDisableAllOtherButtons = () => {
+        const { includeSubfolders } = this.state;
+        return this.numOfSelectedItems() === 1 && includeSubfolders;
+    };
+
+    disableAllOtherItemButtons = itemToKeepId => {
+        const { handleSelectAllTree, items } = this.props;
+
+        if (this.itemsAreLoading()) {
+            return;
+        }
+
+        items.forEach(item => {
+            item.id === itemToKeepId ? (item.isDisabled = false) : (item.isDisabled = true);
+        });
+        this.setState({ allItemsEnabled: false });
+        handleSelectAllTree(itemToKeepId);
+    };
+
+    enableAllItemButtons = () => {
+        const { handleSelectAllTree, items } = this.props;
+        items.forEach(item => {
+            item.isDisabled = false;
+        });
+        this.setState({ allItemsEnabled: true });
+        handleSelectAllTree(null);
+    };
+
+    handleIncludeSubfoldersToggle = () => {
+        const { allItemsEnabled, includeSubfolders, selectedItems } = this.state;
+        const { handleSelectAllTree } = this.props;
+
+        this.setState({ includeSubfolders: !includeSubfolders }, () => {
+            if (this.shouldDisableAllOtherButtons()) {
+                const selectedItem = Object.values(selectedItems)[0];
+                this.setState({ folderToDeepScan: selectedItem }, () => {
+                    if (this.shouldSelectAndActionDisableItems()) {
+                        this.setState({ itemsAreSelectedAndActionDisabled: true });
+                        handleSelectAllTree(selectedItem.id);
+                    } else {
+                        this.disableAllOtherItemButtons(selectedItem.id);
+                    }
+                });
+            } else if (!allItemsEnabled) {
+                this.setState({ folderToDeepScan: {}, itemsAreSelectedAndActionDisabled: false });
+                this.enableAllItemButtons();
+            }
+        });
+    };
+
+    areOtherItemsDisabled = enabledItemId => {
```

Comment:
This and some other functions seem like they could be suited for a helper file. Items can also be passed in as array argument. That way we can write test cases to test them thoroughly with all test cases. 
Name could also be little more specific, perhaps something like `areAllItemsDisabledExcept`?